File: app/src/main/java/com/leo/androidutil/ui/RootActivity.java
package com.leo.androidutil.ui;

import android.os.Bundle;

import androidx.annotation.Nullable;

import android.view.View;
import android.widget.Button;

import com.leo.androidutil.R;
import com.leo.system.IntentUtil;

public class RootActivity extends BaseActivity implements View.OnClickListener {
    private Button mLocationBtn;
    private Button mPinyinBtn;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_root);
        initView();
    }

    private void initView() {
        mLocationBtn = findViewById(R.id.locationBtn);
        mLocationBtn.setOnClickListener(this);
        mPinyinBtn = findViewById(R.id.pinyinBtn);
        mPinyinBtn.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.locationBtn:
                IntentUtil.startActivity(this, LocationActivity.class);
                break;
            case R.id.pinyinBtn:
                IntentUtil.startActivity(this, PinyinActivity.class);
                break;
        }
    }
}

File: app/src/main/java/com/leo/androidutil/ui/RootActivity.kt
package com.leo.androidutil.ui

import android.os.Bundle
import android.view.View
import androidx.databinding.DataBindingUtil
import com.leo.androidutil.R
import com.leo.androidutil.databinding.ActivityRootBinding
import com.leo.system.util.IntentHelper

class RootActivity : BaseActivity(), View.OnClickListener {
    private lateinit var mBinding: ActivityRootBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_root)
        initView()
    }

    private fun initView() {
        mBinding.locationBtn?.setOnClickListener(this)
        mBinding.pinyinBtn?.setOnClickListener(this)
        mBinding.rvDeleteBtn?.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.locationBtn -> IntentHelper.startActivity(this, LocationActivity::class.java)
            R.id.pinyinBtn -> IntentHelper.startActivity(this, PinyinActivity::class.java)
            R.id.rvDeleteBtn -> IntentHelper.startActivity(this, RvDeleteItemActivity::class.java)
        }
    }
}
